plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/bitbucketServerPullRequest.ts

```diff
@@ -0,0 +1,266 @@
+      `${apiBaseUrl}/projects/${project}/repos/${repo}/pull-requests`,
+      data,
+    );
+  } catch (e) {
+    throw new Error(`Unable to create pull-reqeusts, ${e}`);
+  }
+
+  if (response.status !== 201) {
+    throw new Error(
+      `Unable to create pull requests, ${response.status} ${
+        response.statusText
+      }, ${await response.text()}`,
+    );
+  }
+
+  const r = await response.json();
+  return `${r.links.self[0].href}`;
+};
+const findBranches = async (opts: {
+  project: string;
+  repo: string;
+  branchName: string;
+  authorization: string;
+  apiBaseUrl: string;
+}) => {
+  const { project, repo, branchName, authorization, apiBaseUrl } = opts;
+
+  let response: Response;
+  const options: RequestInit = {
+    method: 'GET',
+    headers: {
+      Authorization: authorization,
+      'Content-Type': 'application/json',
+    },
+  };
+
+  try {
+    response = await fetch(
+      `${apiBaseUrl}/projects/${project}/repos/${repo}/branches?boostMatches=true&filterText=${branchName}`,
+      options,
+    );
+  } catch (e) {
+    throw new Error(`Unable to get branches, ${e}`);
+  }
+
+  if (response.status !== 200) {
+    throw new Error(
+      `Unable to get branches, ${response.status} ${
+        response.statusText
+      }, ${await response.text()}`,
+    );
+  }
+
+  const r = await response.json();
+  let branch;
+  for (const object of r.values) {
+    if (object.displayId === branchName) {
+      branch = object;
+      break;
+    }
+  }
+
+  return branch;
+};
+
+/**
+ * Creates a BitbucketServer Pull Request action.
+ * @public
+ */
+export function createPublishBitbucketServerPullRequestAction(options: {
+  integrations: ScmIntegrationRegistry;
+  config: Config;
+}) {
+  const { integrations } = options;
+
+  return createTemplateAction<{
+    repoUrl: string;
+    title: string;
+    description?: string;
+    targetBranch: string;
+    sourceBranch: string;
+    token?: string;
+  }>({
+    id: 'publish:bitbucketServer:pull-request',
+    schema: {
+      input: {
+        type: 'object',
+        required: ['repoUrl'],
+        properties: {
+          repoUrl: {
+            title: 'Repository Location',
+            type: 'string',
+          },
+          title: {
+            title: 'Pull Request Name',
+            type: 'string',
+            description: 'The name for the pull request',
+          },
+          description: {
+            title: 'Pull Request Description',
+            type: 'string',
+            description: 'The description of the pull request',
+          },
+          targetBranch: {
+            title: 'Target Branch',
+            type: 'string',
+            description: 'The description of the pull request',
+          },
+          sourceBranch: {
+            title: 'Source Branch',
+            type: 'string',
+            description: 'The description of the pull request',
+          },
+          token: {
+            title: 'Authentication Token',
```

Comment:
says authentication here, authorization below

Reply:
changed it.